shared/js/toc.jsx

```diff
@@ -26,31 +26,29 @@ loop.shared.toc = (function(mozL10n) {
     propTypes: {
       dispatcher: React.PropTypes.instanceOf(loop.Dispatcher).isRequired,
       isScreenShareActive: React.PropTypes.bool.isRequired,
+      pageStore: React.PropTypes.instanceOf(loop.store.PageStore).isRequired,
       participantStore: React.PropTypes.instanceOf(loop.store.ParticipantStore).isRequired,
       snackbarStore: React.PropTypes.instanceOf(loop.store.SnackbarStore).isRequired
     },
 
     getInitialState: function() {
-      return this.getStoreState();
+      return _.extend(this.getStoreState(), this._getPages());
     },
 
-    componentWillMount: function() {
-      this.onStoreChange(this.getStoreState());
+    _getPages: function() {
+      return {
+        pages: this.props.pageStore.getStoreState("pages")
+      };
     },
 
-    componentWillUpdate: function(nextProps, nextState) {
-      this.onStoreChange(nextState);
+    componentWillMount() {
+      this.props.pageStore.on("change", () => {
+        this.setState(this._getPages());
```

Comment:
Probably won't need to pass down the whole chain of `addTile` anymore as this will listen to events.